ports/stm32/dac.c

```diff
@@ -195,7 +195,15 @@ typedef struct _pyb_dac_obj_t {
     uint8_t outbuf_waveform;
 } pyb_dac_obj_t;
 
-STATIC pyb_dac_obj_t pyb_dac_obj[2];
+void dac_deinit_all(void) {
+    for (int i = 0; i < MP_ARRAY_SIZE(MP_STATE_PORT(pyb_dac_obj_all)); i++) {
```

Comment:
This function can just call `dac_deinit()` on all available channels, no need to check `pyb_dac_obj`.